devito/archinfo.py

```diff
@@ -107,15 +119,16 @@ def get_cpu_brand():
         # Fallback 1: it should now be fine to use psutil
         physical = psutil.cpu_count(logical=False)
         if not physical:
-            # Fallback 2: we might end up here on more exotic platforms such a Power8
-            # Hopefully we can rely on `lscpu`
+            # Fallback 2: we might end up here
+            # on more exotic platforms such a Power8 or due to
+            # erroneous autodetection
             try:
-                physical = lscpu()['Core(s) per socket'] * lscpu()['Socket(s)']
+                physical = (lscpu()['Core(s) per socket'] * lscpu()['Socket(s)'])
             except KeyError:
                 warning("Physical core count autodetection failed")
                 physical = 1
-    cpu_info['physical'] = physical
 
+    cpu_info['physical'] = physical
```

Comment:
this shouldn't have been moved two lines below

Reply:
I think it is one line? Anyway, fixed.